packages/graphql/tests/tck/directives/auth/arguments/roles/roles.test.ts

```diff
@@ -273,24 +279,30 @@ describe("Cypher Auth Roles", () => {
         });
 
         expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
-            "CALL {
-            CREATE (this0:User)
-            SET this0.id = $this0_id
-            SET this0.password = $this0_password
-            WITH this0
-            CALL apoc.util.validate(NOT (any(auth_var1 IN [\\"admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
-            WITH this0
-            CALL apoc.util.validate(NOT (any(auth_var1 IN [\\"super-admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
-            RETURN this0
-            }
-            RETURN [
-            this0 { .id }] AS data"
+            "UNWIND $create_param0 AS create_var1
+            CALL {
+                WITH create_var1
+                CREATE (create_this0:\`User\`)
+                SET
+                    create_this0.id = create_var1.id,
+                    create_this0.password = create_var1.password
+                WITH *
+                CALL apoc.util.validate(NOT (any(auth_var1 IN [\\"admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
+                WITH *
+                CALL apoc.util.validate((create_var1.password IS NOT NULL AND NOT (any(auth_var1 IN [\\"super-admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1)))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
```

Comment:
Intrigued as to where the `create_var1.password IS NOT NULL` has come from given it's not in the original check from line 283?

Reply:
So, this is a change needed when we are in the unwind optimization context when `auth` is applied at the field level, the previous behavior was to include `auth` only to the subquery where the field was present in the input argument. As of now, we have only one `Subquery` the `create_var1.password IS NOT NULL` is added to simulate the previous, in this way the auth validation could return `true` only if `password` is present in the input for that particular item!